tests/integration/vxlan_test.py

```diff
@@ -0,0 +1,179 @@
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+import time
+
+import pytest
+
+import libnmstate
+from libnmstate.error import NmstateVerificationError
+from libnmstate.schema import VXLAN
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+
+from .testlib import assertlib
+from .testlib import statelib
+from .testlib.vxlan import vxlan_interface
+
+VXLAN1_ID = 101
+VXLAN2_ID = 102
+VXLAN1_IFNAME = 'eth1.{}'.format(VXLAN1_ID)
+VXLAN2_IFNAME = 'eth1.{}'.format(VXLAN2_ID)
+
+TWO_VXLANS_STATE = {
+    Interface.KEY: [
+        {
+            'name': VXLAN1_IFNAME,
+            'type': 'vxlan',
+            'state': 'up',
+            VXLAN.CONFIG_SUBTREE: {
+                VXLAN.ID: VXLAN1_ID,
+                VXLAN.BASE_IFACE: 'eth1',
+                VXLAN.REMOTE: '192.168.100.1',
+            },
+        },
+        {
+            'name': VXLAN2_IFNAME,
+            'type': 'vxlan',
+            'state': 'up',
+            VXLAN.CONFIG_SUBTREE: {
+                VXLAN.ID: VXLAN2_ID,
+                VXLAN.BASE_IFACE: 'eth1',
+                VXLAN.REMOTE: '192.168.100.2',
+            },
+        },
+    ]
+}
+
+
+def test_add_and_remove_vxlan(eth1_up):
+    with vxlan_interface(
+        VXLAN1_IFNAME, VXLAN1_ID, eth1_up[Interface.KEY][0][Interface.NAME]
+    ) as desired_state:
+        assertlib.assert_state(desired_state)
+
+    current_state = statelib.show_only((VXLAN1_IFNAME,))
+    assert not current_state[Interface.KEY]
+
+
+@pytest.fixture
+def vxlan_on_eth1(eth1_up):
```

Comment:
nit: The convention in our tests is to place the fixtures at the top, before the tests.

It seems no one uses it.

But if you intend to use it, please do not specify the name of the base in the fixture name... we can abstract it by not mentioning it at all. Calling it `vxlan0_up` should be enough.